packages/backup/src/Helper_Script_Manager.php

```diff
@@ -0,0 +1,322 @@
+<?php
+/**
+ * The Jetpack Backup Helper Script Manager class.
+ *
+ * @package automattic/jetpack-backup
+ */
+
+namespace Automattic\Jetpack\Backup;
+
+/**
+ * Helper_Script_Manager manages installation, deletion and cleanup of Helper Scripts
+ * to assist with backing up Jetpack Sites.
+ */
+class Helper_Script_Manager {
+
+	const RELATIVE_INSTALL_LOCATIONS = array( '', 'wp-content', 'wp-content/uploads' );
+	const TEMP_DIRECTORY             = 'jetpack-temp';
+	const HELPER_HEADER              = "<?php /* Jetpack Backup Helper Script */\n";
+	const EXPIRY_TIME                = 8 * 3600; // 8 hours
+	const MAX_FILESIZE               = 1024 * 1024; // 1 MiB
+
+	const README_LINES = array(
+		'These files have been put on your server by Jetpack to assist with backups and restores of your site content. They are cleaned up automatically when we no longer need them.',
+		'If you no longer have Jetpack connected to your site, you can delete them manually.',
+		'If you have questions or need assistance, please contact Jetpack Support at https://jetpack.com/support/',
+		'If you like to build amazing things with WordPress, you should visit automattic.com/jobs and apply to join the fun – mention this file when you apply!;',
+	);
+
+	const INDEX_FILE = '<?php // Silence is golden';
+
+	/**
+	 * Installs a Helper Script, and returns its filesystem path and access url.
+	 *
+	 * @access public
+	 * @static
+	 *
+	 * @param string $script_body Helper Script file contents.
+	 * @return array|WP_Error     Either an array containing the path and url of the helper script, or an error.
+	 */
+	public static function install_helper_script( $script_body ) {
+		// Check that the script body contains the correct header.
+		if ( strncmp( $script_body, self::HELPER_HEADER, strlen( self::HELPER_HEADER ) ) !== 0 ) {
+			return new \WP_Error( 'invalid_helper', 'Invalid Helper Script header' );
+		}
+
+		// Refuse to install a Helper Script that is too large.
+		if ( strlen( $script_body ) > self::MAX_FILESIZE ) {
+			return new \WP_Error( 'invalid_helper', 'Invalid Helper Script size' );
+		}
+
+		// Create a jetpack-temp directory for the Helper Script.
+		$relative_temp_dir = self::create_temp_directory();
+		if ( is_wp_error( $relative_temp_dir ) ) {
+			return $relative_temp_dir;
+		}
+
+		// Generate a random filename, avoid clashes.
+		$max_attempts = 5;
+		for ( $attempt = 0; $attempt < $max_attempts; $attempt++ ) {
+			$file_key           = wp_generate_password( 10, false );
+			$relative_file_path = trailingslashit( $relative_temp_dir ) . 'jp-helper-' . $file_key . '.php';
+			$absolute_file_path = trailingslashit( ABSPATH ) . $relative_file_path;
```

Comment:
Should this be `\ABSPATH` because we're in a different namespace from the global one here?

Reply:
Yes, probably better to be explicit. This instance of ABSPATH was refactored out based on previous feedback just as your comments were added -- but I've added \\ to the new instance of ABSPATH (and similar constants) in the latest version. 

Thanks for the feedback! :)

Reply:
Thank you! Also, noting that we should be able to use the `constants` package to safely check and retrieve constants.